GoMeet/src/main/webapp/js/meeting-location-dao.js

```diff
@@ -1,26 +1,128 @@
+/** Constructs a Data Access Object for location data. */
+class MeetingLocationDaoFactory {
+  /** 
+   * Returns a meetingLocationDAO object based on whether the map stores to a temporary
+   * or permanent location.
+   */
+  static getLocationDao(storageType) {
+    if (storageType === 'permanent') {
+      return new PermMeetingLocationDao();
+    } else {
+      return new TempMeetingLocationDao();
+    }
+  }
+}
+
+/** Stores location data to session storage. */
+class TempMeetingLocationDao {
+  constructor() {
+    this.emptyKeyString = '';
+    this.MAX_NUM_ENTITIES = 5;
+  }
+
+  /** Fetches the location data stored in session storage. */
+  fetchLocations() {
+    if (sessionStorage.getItem('locations') == null) {
+      return [];
+    } else {
+      return JSON.parse(sessionStorage.getItem('locations'));
+    } 
+  }
+
+  /** Saves a new location to session storage. */
+  newLocation(title, lat, lng, note) {
+    // Check if the new location can be added.
+    if (title === '') {
+      throw new Error(BLANK_FIELDS_ALERT);
+    }
+    if (!this.validTitle(title)) {
+      throw new Error(SAME_TITLE);
+    }
+    if (this.maxEntitiesReached()) {
+      throw new Error(MAX_ENTITIES);
+    }
+
+    // Add the new location.
+    let locationList;
+    if (sessionStorage.getItem('locations') == null) {
+      locationList = [];
+    } else {
+      locationList = JSON.parse(sessionStorage.getItem('locations'));
+    }
+    const newLocation = {title: title, lat: lat, lng: lng,
+        note: note, keyString: this.emptyKeyString};
+    locationList.push(newLocation);
+    sessionStorage.setItem('locations', JSON.stringify(locationList));
+  }
+
+  /**
+   * Called when the user clicks the VOTE button.
+   * Note: Temporary storage is only used during the meeting creation.
+   * The meeting creator cannot vote more than once.
+   * @throws an error when the meeting creator tries to vote for location.
+   */
+  updateLocation(keyString) {
+    throw new Error(USER_HAS_VOTED);
```

Comment:
So the meeting creator can't vote for the locations that they proposed?

Reply:
Yep, when the location is created it automatically has one vote, which is the meeting creator's.